plugins/janus_sip.c

```diff
@@ -1937,7 +1958,7 @@ int janus_sip_init(janus_callbacks *callback, const char *config_path) {
 	sessions = g_hash_table_new_full(NULL, NULL, NULL, (GDestroyNotify)janus_sip_session_destroy);
 	identities = g_hash_table_new(g_str_hash, g_str_equal);
 	callids = g_hash_table_new(g_str_hash, g_str_equal);
-	messageids = g_hash_table_new(g_str_hash, g_str_equal);
+	messageids = g_hash_table_new_full(NULL, NULL, (GDestroyNotify)janus_sip_message_key_destroy, (GDestroyNotify)janus_sip_session_destroy);
```

Comment:
No, this should not be `janus_sip_session_destroy`: this will mark the whole session as destroyed, and prevent the SIP stack to be used further. Besides, it will cause leaks, because `janus_sip_session_destroy` is already used as the callback to use for actually getting rid of a session, and since it can only be invoked once, a reference will remain dangling. You should instead add a new function that just dereferences a session, e.g., like we do here for publishers in the VideoRoom:

https://github.com/meetecho/janus-gateway/blob/master/plugins/janus_videoroom.c#L1750

so basically something like a `janus_sip_session_dereference` function that you can use as a `GDestroyNotify` callback to remove the reference you added previously.

Reply:
ack